packages/tenant-process/src/services/tenantService.ts

```diff
@@ -515,6 +519,97 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async revokeVerifiedAttribute(
+      {
+        tenantId,
+        attributeId,
+        revokerId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        attributeId: AttributeId;
+        revokerId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(
+        `Revoke verified attribute ${attributeId} to tenant ${tenantId}`
+      );
+
+      if (revokerId === tenantId) {
+        throw verifiedAttributeSelfRevocationNotAllowed();
+      }
+
+      const allowedStatuses = [
+        agreementState.pending,
+        agreementState.active,
+        agreementState.suspended,
+      ];
+
+      await assertVerifiedAttributeOperationAllowed({
+        producerId: revokerId,
+        consumerId: tenantId,
+        attributeId,
+        agreementStates: allowedStatuses,
+        readModelService,
+        error: attributeRevocationNotAllowed(tenantId, attributeId),
+      });
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const verifiedTenantAttribute = targetTenant.data.attributes.find(
+        (attr): attr is VerifiedTenantAttribute =>
+          attr.type === tenantAttributeType.VERIFIED && attr.id === attributeId
+      );
+
+      if (!verifiedTenantAttribute) {
+        throw attributeNotFound(attributeId);
+      }
+
+      const verifier = verifiedTenantAttribute.verifiedBy.find(
+        (a) => a.id === revokerId
+      );
+
+      if (!verifier) {
+        throw attributeRevocationNotAllowed(tenantId, attributeId);
+      }
+
+      if (verifiedTenantAttribute.revokedBy.find((a) => a.id === revokerId)) {
+        throw attributeAlreadyRevoked(tenantId, revokerId, attributeId);
+      }
+
+      const updatedTenant = {
+        ...targetTenant.data,
+        attributes: targetTenant.data.attributes.map((attr) =>
+          attr.id === attributeId
+            ? ({
+                ...verifiedTenantAttribute,
+                verifiedBy: verifiedTenantAttribute.verifiedBy.filter(
+                  (v) => v.id !== revokerId
+                ),
```

Comment:
I am not sure about also removing the revoker from the ones who previously verified that attribute.
cc @galales

Reply:
> I am not sure about also removing the revoker from the ones who previously verified that attribute. cc @galales

It should be correct since It's the same behaviour as [in Scala](https://github.com/pagopa/interop-be-tenant-process/blob/7c6596aa4665ed333915765c6d818d9614554cbe/src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala#L968)

Reply:
Understood